networkx/algorithms/tests/test_matching.py

```diff
@@ -450,7 +450,7 @@ def test_selfloops(self):
         assert not nx.is_matching(G, {(0, 0), (1, 2), (2, 3)})
         # selfloop edge in G
         G.add_edge(0, 0)
-        assert not nx.is_matching(G, {(0, 0), (1, 2), (2, 3)})
+        assert not nx.is_matching(G, {(0, 0), (1, 2)})
```

Comment:
Very nice catch - as originally written, the test was not hitting the intended branch!